sdk/clientcore/http-okhttp3/src/main/java/io/clientcore/http/okhttp3/implementation/ProxyAuthenticator.java

```diff
@@ -84,76 +95,90 @@ public Interceptor getProxyAuthenticationInfoInterceptor() {
      */
     @Override
     public Request authenticate(Route route, Response response) {
-        String authorizationHeader =
-            challengeHandler.attemptToPipelineAuthorization(PROXY_METHOD, PROXY_URI_PATH, NO_BODY);
-
-        // Pipelining was successful, use the generated authorization header.
-        if (!isNullOrEmpty(authorizationHeader)) {
-            return response.request().newBuilder()
-                .header(PROXY_AUTHORIZATION, authorizationHeader)
-                .build();
-        }
-
-        // If this is a pre-emptive challenge quit now if pipelining doesn't produce anything.
+        // Handle preemptive authentication
         if (PREEMPTIVE_AUTHENTICATE.equalsIgnoreCase(response.message())) {
+            // If we have already authenticated, apply the stored Proxy-Authorization header.
+            if (lastAuthorizationHeader != null) {
```

Comment:
Should make the lastAuthorizationHeader volatile, this could be used with multiple threads.